ion/src/main/java/com/fasterxml/jackson/dataformat/ion/IonParser.java

```diff
@@ -286,6 +286,13 @@ public String getText() throws IOException
                         msg = "UNKNOWN ROOT CAUSE";
                     }
                     throw _constructError("Internal `IonReader` error: "+msg, e);
+                } catch (NullPointerException e) {
```

Comment:
I think this could then be combined with above as

    } catch (AssertionError | NullPointerException e) {

(or whatever the syntax was to allow multiple exception types)
since handling is basically the same, for now